wrangler-core/src/main/java/io/cdap/wrangler/expression/ELContext.java

```diff
@@ -36,6 +38,23 @@
 public class ELContext implements JexlContext {
   private final Map<String, Object> values = new HashMap<>();
 
+  /**
+   * Context object passed to every expression evaluation.
+   * All properties of this class are public to ensure they can be accessed with dot(.) format.
+   */
+  public static class Context {
+    public String environment;
+    public String name;
+    public long nano;
+    public long millis;
```

Comment:
don't think it makes sense for an expression to modify any of these right? If so, make these final otherwise directives can mess each other up.

Reply:
done!